operator/v1/types.go

```diff
@@ -52,6 +52,43 @@ type ClusterManagerSpec struct {
 	// +optional
 	// +kubebuilder:default={mode: Default}
 	DeployOption DeployOption `json:"deployOption,omitempty"`
+
+	// Detached includes configurations we needs for clustermanager in the detached mode.
+	// +optional
+	Detached DetachedClusterManagerConfiguration `json:"detached,omitempty"`
+}
+
+// DetachedClusterManagerConfiguration represents customized configurations we need to set for clustermanager in the detached mode.
+// After a detached mode deployment, the webhook-servers will not be feasible until the user exposes the webhook-servers in the management cluster manually. And the way to do that is depending on the platform and user's preference.
+// For example, if we are using a OCP platform, we could use Route to expose the webhook-servers.
+type DetachedClusterManagerConfiguration struct {
+	// RegistrationWebhookConfiguration represents the customized webhook-server configuration of registration.
+	// +optional
+	RegistrationWebhookConfiguration WebhookConfiguration `json:"registrationWebhookConfiguration,omitempty"`
+
+	// WorkWebhookConfiguration represents the customized webhook-server configuration of work.
+	// +optional
+	WorkWebhookConfiguration WebhookConfiguration `json:"workWebhookConfiguration,omitempty"`
+}
+
+// WebhookConfiguration has two properties: Address and Port.
+// The registration and work provide webhook-server by using [generic-admission-server](https://github.com/openshift/generic-admission-server).
```

Comment:
I do not think this doc needs to be added. These are internal implementation

Reply:
Yes, removed this part.